features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Resources/components/operationComponents/parameterUtils.js

```diff
@@ -0,0 +1,108 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+/**
+ * Defining the supported data types by the OAS spec versions.
+ *
+ * 2.0 : https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#dataTypeFormat
+ * 3.x :https://swagger.io/docs/specification/data-models/data-types/
+ * */
+const supportedDataTypes = {
+    '2.0': {
+        body: ['integer', 'number', 'string', 'boolean', 'object'],
+        formData: ['integer', 'number', 'string', 'boolean', 'file'],
+        query: ['integer', 'number', 'string', 'boolean'],
+        path: ['integer', 'number', 'string', 'boolean'],
+        header: ['integer', 'number', 'string', 'boolean'],
+    },
+    '3.x': {
+        body: ['integer', 'number', 'string', 'boolean', 'object'],
+        query: ['integer', 'number', 'string', 'boolean'],
+        path: ['integer', 'number', 'string', 'boolean'],
+        header: ['integer', 'number', 'string', 'boolean'],
+        cookie: ['integer', 'number', 'string', 'boolean'],
+    },
+};
+
+/**
+ * Util method for checking conditions.
+ *
+ * @param {boolean} condition : The condition to be checked
+ * @param {any} matched : The value to be returned if the condition is true
+ * @param {any} otherwise : The value to be returned if the condition is false.
+ * @return {object} If true -> matched, false -> otherwise
+ * */
+export function iff(condition, matched, otherwise) {
```

Comment:
@menakaj Why we need a util function for this conditional ternary operator ? is it to get away with eslint warning ?